app/api/account/route.js

```diff
@@ -0,0 +1,36 @@
+import { NextResponse } from 'next/server';
+import connect from "../../../src/utils/data/db";
+import User from "../../../src/models/User";
+import Tasks from "../../../src/models/Tasks";
+
+//retrieve user info
+export const GET = async () => {
+    await connect();
+    const user = await User.findOne({ username: 'johndoe123' });
+
+    try {
+        return NextResponse.json(user);
+    } catch (error) {
+        console.error(error);
+        return NextResponse.json({ message: 'Internal Server Error' }, {status: 500});
+  }
+}
+
+//save new username
+export const POST = async (req) => {
+    await connect();
+
+    const result = await req.json();
+    const newUsername = result.inputValue;
+
+    const filter = { username: 'johndoe123' };
```

Comment:
Filter should be passed through http request body

Reply:
Maybe edit it after login function is done, then we can use session or cookie to do this?

Reply:
Yes, I'll edit it in the next version.